daemon/containerd/image_exporter.go

```diff
@@ -342,3 +378,62 @@ func (i *ImageService) LoadImage(ctx context.Context, inTar io.ReadCloser, platf
 
 	return nil
 }
+
+// verifyImagesProvidePlatform checks if the requested platform is loaded.
+// If the requested platform is not loaded, it returns an error.
+func (i *ImageService) verifyImagesProvidePlatform(ctx context.Context, imgs []containerdimages.Image, platform ocispec.Platform, pm platforms.Matcher) error {
+	var imageNames []string
+	for _, img := range imgs {
+		hasRequestedPlatform := false
+		err := i.walkImageManifests(ctx, img, func(platformImg *ImageManifest) error {
+			imgPlat, err := platformImg.ImagePlatform(ctx)
+			if err != nil {
+				if cerrdefs.IsNotFound(err) {
+					return nil
+				}
+				return errors.Wrapf(err, "failed to determine image platform")
+			}
+
+			if pm.Match(imgPlat) {
+				available, err := platformImg.CheckContentAvailable(ctx)
+				if err != nil {
```

Comment:
Note to self (and others): I checked  `platformImg.CheckContentAvailable(ctx)`, and it only returns an error on failures, not on "missing" (i.e. `NotFound` errors are not returned)